scripts/trivia.coffee

```diff
@@ -39,39 +64,79 @@ module.exports = (robot) ->
           msg.send solutionHint
       , 15000
 
-  #function to turn off trivia
-  #made this so i could reset every time a question is done
-  triviaStop = (msg) ->
-    clearInterval(hintInterval)
-    hintInterval = null
-    triviaOn = false
-    
-  #solve attempt matches anything after !, case insensitive
-  #need to add some way to score points
-  robot.respond /!(.+)$/i, (msg) ->
-    solveAttempt = msg.match[1]
+  #-------------Solve Attempt-----------------#
+  #solve attempt matches anything after ?  
+  robot.respond /\?\s*(.+)$/i, (msg) ->
+    solveAttempt = msg.match[1].trim()
     if solveAttempt.toLowerCase() == currentSolution.toLowerCase()
       msg.reply "Ding Ding Ding, you are correct!"
+      
+      #get user and add 1 point to their score
+      username = msg.message.user.name
+      points[username] ?= 0
+      points[username] += 1
+
+      #update points object in db
+      robot.brain.data.points = points
+
+      msg.reply "You currently have " + points[username] + " points"
+
       triviaStop(msg)
       triviaStart(msg)
     else
       msg.reply "Incorrect!"
-
-  #command to turn off trivia
-  robot.respond /trivia off/i, (msg) ->
+  
+  #---------------Trivia Toggling---------------#
+  #used for setting new question and turning off trivia
+  triviaStop = (msg) ->
+    clearInterval(hintInterval)
+    hintInterval = null
+    triviaOn = false
+  
+  #turn off trivia
+  robot.respond /(trivia\s)?off/i, (msg) ->
     if triviaOn
       triviaStop(msg) 
       msg.send "Trivia is now off"
     else msg.send "Trivia is not in progress"
 
+  #turn on trivia
   robot.respond /trivia(\son)?$/i, (msg) ->
     if !triviaOn
       msg.send "Trivia started"
       triviaStart(msg)
     else msg.send "Trivia already in progress"
 
+  #----------------Leaderboard----------------#
+  #show top 10 on leaderboard + their points
   robot.respond /leaderboard/i, (msg) ->
-    msg.send "Currently there is no database set up to keep track of scores, feel free to contribute!"
+    sortable = []
+    for user,pts of robot.brain.data.points
+      sortable.push([user,pts])
+    sorted = sortable.sort((a,b) ->
+      b[1]-a[1]
+    )
+    i = 0
+    for user in sorted
+      if i==10
+        break
+      i++
```

Comment:
this whole thing can be done in one line https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice